languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10083,6 +10083,55 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+
+      <rulegroup id="PLACE_DE_LA_VIRGULE" name="place de la virgule" default="temp_off">
+          <rule>
+              <pattern>
+                  <marker>
+                      <token postag="[RAVNJPZCD].*" postag_regexp="yes"/>
+                      <token>c'</token>
+                      <token>est</token>
+                  </marker>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2\3</suggestion>
+              <example correction="grand, c'est">Le plus <marker>grand c'est</marker> Pierre.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token postag="N.*" postag_regexp="yes"/>
+                      <token regexp="yes">ni|mais|car|or|c’est-à-dire|soit|voire|à savoir</token>
+                  </marker>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2</suggestion>
+              <example correction="temps, mais">Il ne réagit pas au <marker>temps mais</marker> au lieu.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token postag="V.*" postag_regexp="yes"/>
+                      <token postag="V.*" postag_regexp="yes"/>
+                      <token postag="R pers suj.*" postag_regexp="yes"/>
+                  </marker>
+              </pattern>
+              <message>Des virgules semblent être requises dans ce contexte.</message>
+              <suggestion>\1, \2\3,</suggestion>
+              <example correction="pense, dit-elle,">Il le <marker>pense dit-elle</marker> mais ce n'est pas vrai.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token postag="N.*" postag_regexp="yes"/>
+                      <token postag="R pers suj.*" postag_regexp="yes"/>
+                  </marker>
+              </pattern>
+              <message>Une virgule semble être requise dans ce contexte.</message>
+              <suggestion>\1, \2</suggestion>
+              <example correction="photos, je">Ces <marker>photos je</marker> les aime beaucoup.</example>
+          </rule>
```

Comment:
Same here: please look at the false positives in the rule editor results and make exceptions/antipatterns as needed. Keep in mind: the goal is to make it find as many errors as possible. So make your antipatterns/exceptions as specific as possible.

Example:

> C'est un cairn **car il** est constitué de pierres sèches.

Here we go with the coach again ;) to avoid this, you could add an exception to the first token:
`<token postag="N.*" postag_regexp="yes"><exception postag="C.*" postag_regexp="yes"/></token>`

Please look at the sentences the rule editor spits out for each of these subrules. If there are false positives among those sentences, make them all disappear (with the real errors staying there, if possible).